### PR TITLE
MES-2336 - Add Journal Data to the state 

### PR DESCRIPTION
**Pull request by Cavalier07**
## Description and relevant Jira numbers

- Updates the state so any data from the journal is nested under a Journal Data Attribute
- Updated every page which uses anything the above data still works
- Renamed Test Centre data selector to getTestCentre from getJournalData

## Pull Request checklist:

- [x] [WIP] tag removed from PR title
- [x] PR has an understandable description

## Git feature branch checklist:

- [x] branch name comply with our branching strategy
- [x] git branch contains relevant JIRA ticket number
- [x] branch rebased against the latest develop
- [ ] commits are squashed

## Sign off process checklist:

- [ ] Code has been tested manually
- [ ] Tests are passing
- [ ] PR link added to JIRA
- [ ] Reviewers selected in Github
- [x] Any new reusable component/widget added to component library on Confluence

- [x] Screenshot(s) captured on the physical device (if applicable)
- [ ] Tested by QA
- [ ] PO's approval

### CODE REVIEW

File: src/modules/tests/journal-data/journal-data.reducer.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+      return {
+        test: 12345,
+        ...combineReducers({
+          examiner: examinerReducer,
+          testCentre: testCentreReducer,
+          testSlotAttributes: testSlotsAttributesReducer,
+          candidate: candidateReducer,
+          applicationReference: applicationReferenceReducer,
+        })(state, action),
+      };
```
Comment (Cavalier07) [inline src/modules/tests/journal-data/journal-data.reducer.ts]: Have tried: 

- combine reducers, 
- calling the reducers manually, 
- trying to do nesting tests.ts in

All end up in the same result and break the reducer/state tree

File: src/modules/tests/test-centre/test-centre.reducer.ts [hunk 2]
```diff
@@ -17,4 +17,4 @@ export function testCentreReducer(
   return state;
 }
 
-export const getJournalData = createFeatureSelector<TestCentre>('testCentre');
+export const getTestCentre = createFeatureSelector<TestCentre>('testCentre');
```
Comment (johnmorleybjss) [inline src/modules/tests/test-centre/test-centre.reducer.ts]: 😅

File: src/pages/health-declaration/health-declaration.ts [hunk 3]
```diff
@@ -131,18 +131,21 @@ export class HealthDeclarationPage extends BasePageComponent {
       candidateName$: this.store$.pipe(
         select(getTests),
         select(getCurrentTest),
+        select(getJournalData),
```
Comment (szabolcs-keresztes) [inline src/pages/health-declaration/health-declaration.ts]: Worth thinking about extracting `journalData$` for simplified code

Comment (Cavalier07) [reply src/pages/health-declaration/health-declaration.ts]: In the future yes - for now I would say it's useful there to let people know there is another layer in the app

Comment (szabolcs-keresztes) [reply src/pages/health-declaration/health-declaration.ts]: Could have reused the `currentTest$` observable at least

**Pull request merged**